controllers/hostingdeploymentautoscalingjob/hostingdeploymentautoscalingjob_controller_test.go

```diff
@@ -0,0 +1,762 @@
+				ScalableTargets: scalableTargets,
+			}).
+			AddDescribeScalingPoliciesResponse(applicationautoscaling.DescribeScalingPoliciesOutput{
+				ScalingPolicies: scalingPolicies,
+			}).
+			Build()
+
+		controller := createReconciler(k8sClient, applicationAutoscalingClient)
+		request := CreateReconciliationRequest(hostingdeploymentautoscalingjob.ObjectMeta.Name, hostingdeploymentautoscalingjob.ObjectMeta.Namespace)
+
+		controller.Reconcile(request)
+
+		Expect(receivedRequests.Len()).To(Equal(6))
+	})
+
+	It("should add the finalizer and update the status", func() {
+		policyName := "test-policy-name"
+		policyArn := "policy-arn"
+		resourceId := "endpoint/endpoint-xyz/variant/variant-xyz"
+		scalableTarget := applicationautoscaling.ScalableTarget{
+			ResourceId: &resourceId,
+		}
+		scalableTargets := []applicationautoscaling.ScalableTarget{scalableTarget}
+		scalingPolicy := applicationautoscaling.ScalingPolicy{
+			PolicyName: &policyName,
+			PolicyARN:  &policyArn,
+			ResourceId: &resourceId,
+		}
+		scalingPolicies := []applicationautoscaling.ScalingPolicy{scalingPolicy}
+
+		applicationAutoscalingClient := mockAutoscalingClientBuilder.
+			AddDescribeScalableTargetsResponse(applicationautoscaling.DescribeScalableTargetsOutput{}).
+			AddDescribeScalingPoliciesResponse(applicationautoscaling.DescribeScalingPoliciesOutput{
+				ScalingPolicies: []applicationautoscaling.ScalingPolicy{},
+			}).
+			AddRegisterScalableTargetsResponse(applicationautoscaling.RegisterScalableTargetOutput{}).
+			AddPutScalingPolicyResponse(applicationautoscaling.PutScalingPolicyOutput{
+				PolicyARN: &policyArn,
+			}).
+			AddDescribeScalableTargetsResponse(applicationautoscaling.DescribeScalableTargetsOutput{
+				ScalableTargets: scalableTargets,
+			}).
+			AddDescribeScalingPoliciesResponse(applicationautoscaling.DescribeScalingPoliciesOutput{
+				ScalingPolicies: scalingPolicies,
+			}).
+			Build()
+
+		controller := createReconciler(k8sClient, applicationAutoscalingClient)
+		request := CreateReconciliationRequest(hostingdeploymentautoscalingjob.ObjectMeta.Name, hostingdeploymentautoscalingjob.ObjectMeta.Namespace)
+
+		controller.Reconcile(request)
+
+		err := k8sClient.Get(context.Background(), types.NamespacedName{
+			Namespace: hostingdeploymentautoscalingjob.ObjectMeta.Namespace,
+			Name:      hostingdeploymentautoscalingjob.ObjectMeta.Name,
+		}, hostingdeploymentautoscalingjob)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Verify a finalizer has been added
+		Expect(hostingdeploymentautoscalingjob.ObjectMeta.GetFinalizers()).To(ContainElement(SageMakerResourceFinalizerName))
+
+		Expect(receivedRequests.Len()).To(Equal(6))
+		Expect(hostingdeploymentautoscalingjob.Status.HostingDeploymentAutoscalingJobStatus).To(Equal(CreatedAutoscalingJobStatus))
+		Expect(hostingdeploymentautoscalingjob.Status.PolicyName).To(Equal(policyName))
+		// TODO mbaijal: Add PolicyARN ?
```

Comment:
Is this tested elsewhere as well as just here?

Reply:
I actually have not needed to track the policyARN anywhere - there is a TODO to add it to the status but it adds little value except show up in logs.